dll.py

```diff
@@ -0,0 +1,40 @@
+class Node(object):
+
+    def __init__(self, vertex, prev, next):
+        self.vertex = vertex
+        self.prev = prev
+        self.next = next
+
+
+class DoubleLinkedList(object):
+    firstNode = None
+    lastNode = None
+
+    def insert_after(self, node, new_node):
+        new_node.prev = node
+        if node.next is None:
+            new_node.next = None
+            self.lastNode = new_node
+        else:
+            new_node.next = node.next
+            node.next.prev = new_node
+        node.next = new_node
+
+    def insert_before(self, node, new_node):
+        new_node.next = node
+        if node.prev is None:
+            new_node.prev = None
+            self.firstNode = new_node
+        else:
+            new_node.prev = node.prev
+            node.prev.next = new_node
+        node.prev = new_node
+
+    def append(self, node):
+        if self.firstNode is None:
+            self.firstNode = node
+            self.lastNode = node
+            node.prev = None
+            node.next = None
+        else:
+            self.insert_after(self.lastNode, node)
```

Comment:
Ik mis ook nog remove en pop methodes.

Reply:
Zie bv https://gist.github.com/nichochar/87e18f9eb72f114853eb

Reply:
Remove moet inderdaad nog, maar wat doen push en pop, want normaal gesproken gaat dat over een stack en dan is het volgens mij het eerste element. Wil je dat hier ook? of het laatste element? Of maakt het je niet uit?

Reply:
pop doet poplast, en push meestal pushleft (en append pushright volgens mij).
De kant maakt me niet zoveel uit (al zal het ene vast efficienter zijn dan de andere).
Apparte functies zijn wel handig (tov insert_after, insert_before), omdat deze aangewezen efficient zijn (en vermoedelijk heeft meest gebruikt zullen worden).

We moeten hem mogelijk ook nog iterable maken (zodat je `for i in dll` kunt doen). https://docs.python.org/3/tutorial/classes.html#iterators